contracts/consensus/Consensus.sol

```diff
@@ -1178,32 +1178,33 @@ contract Consensus is MasterCopyNonUpgradable, CoreLifetimeEnum, MosaicVersion,
 
     /**
      * @notice hashBlockSegment() function calculates a seed based on the
-     *         given sengment (start, end].
+     *         specified blocksengment.
      *
-     * @param start Start block number (not included in the segment).
-     * @param end End block number (included in the segment).
+     * @param end The ending block number of the blocksegment (included).
+     * @param length Length of the blocksegment.
      *
      * @return Returns a seed based on the blockhashes of the given blocksegment.
      */
     function hashBlockSegment(
-        uint256 start,
-        uint256 end
+        uint256 end,
+        uint256 length
     )
         private
         view
         returns (bytes32 seed_)
     {
+        assert(length > 0);
+
+        uint256 begin = end.add(uint256(1)).sub(length);
```

Comment:
1. `hashBlockSegment` is used only at one place, so we need to pass the length, its already a constant.
2. For me, It was difficult to understand this part of the code, just a suggestion.
Instead of adding `add(uint256(1))` here, can be do the following in the for loop: 

```
for (uint256 i = 1; i <= length; i = i.add(1)) {
            seedGenerator[i] = blockhash(begin.add(i));
        }	        
```

Reply:
1. assuming you meant "we don't" need to pass the length? agreeing with this comment

Reply:
@deepesh-kn with regards to the 2. point I would rather keep existing implementation.
@deepesh-kn @benjaminbollen wrt 1. point, I am not sure why should not be function written as a standalone function. However, I will change it as you both have proposed.

Reply:
Done.

Reply:
> 1. assuming you meant "we don't" need to pass the length? agreeing with this comment

yes, I meant `we don't`.